File: backend/UserService/DF.UserService.Contracts/Models/Response/BusinessDashboardResponse.cs
namespace DF.UserService.Contracts.Models.Response;

/// <summary>
/// Aggregate financial dashboard for a business, computed live from the
/// business's Stripe Connect account. All monetary fields are in the
/// account's settlement currency (typically USD) as decimals already
/// converted from Stripe's minor-unit integers.
/// </summary>
public sealed record BusinessDashboardResponse(
    BusinessDashboardSummary Summary,
    IReadOnlyList<DashboardTimePoint> TimeSeries,
    IReadOnlyList<DashboardBreakdownItem> OutcomeBreakdown,
    IReadOnlyList<DashboardPayout> Payouts,
    BusinessBalanceSnapshot Balance,
    DashboardWindow Window,
    string Currency
);

public sealed record BusinessDashboardSummary(
    decimal GrossSales,
    decimal Refunds,
    decimal PlatformFees,
    decimal StripeProcessingFees,
    decimal NetIncome,
    decimal PaidOut,
    int Orders,
    int RefundedOrders
);

public sealed record DashboardTimePoint(
    DateTime Date,        // UTC midnight of the day
    decimal Gross,
    decimal Refunds,
    decimal Fees,
    decimal Net
);

public sealed record DashboardBreakdownItem(
    string Category,      // "Net income" | "Platform fee" | "Stripe fees" | "Refunds"
    decimal Amount
);

public sealed record DashboardPayout(
    string Id,
    DateTime CreatedAtUtc,
    DateTime? ArrivalUtc,
    DateTime? PaidAtUtc,
    decimal Amount,
    string Currency,
    string Status,         // pending/in_transit/paid/failed/canceled
    string? FailureMessage
);

public sealed record BusinessBalanceSnapshot(
    decimal Available,
    decimal Pending,
    string Currency
);

public sealed record DashboardWindow(
    DateTime FromUtc,
    DateTime ToUtc
);

File: frontend/food-delivery-platform/src/models/responses/business-dashboard/BusinessDashboardResponse.ts
export interface BusinessDashboardSummary {
    grossSales: number
    refunds: number
    platformFees: number
    stripeProcessingFees: number
    netIncome: number
    paidOut: number
    orders: number
    refundedOrders: number
}

export interface DashboardTimePoint {
    date: string         // ISO date
    gross: number
    refunds: number
    fees: number
    net: number
}

export interface DashboardBreakdownItem {
    category: string
    amount: number
}

export interface DashboardPayout {
    id: string
    createdAtUtc: string
    arrivalUtc?: string | null
    paidAtUtc?: string | null
    amount: number
    currency: string
    status: string
    failureMessage?: string | null
}

export interface BusinessBalanceSnapshot {
    available: number
    pending: number
    currency: string
}

export interface DashboardWindow {
    fromUtc: string
    toUtc: string
}

export interface BusinessDashboardResponse {
    summary: BusinessDashboardSummary
    timeSeries: DashboardTimePoint[]
    outcomeBreakdown: DashboardBreakdownItem[]
    payouts: DashboardPayout[]
    balance: BusinessBalanceSnapshot
    window: DashboardWindow
    currency: string
}

export interface DishRevenue {
    dishId: string
    dishName: string
    quantitySold: number
    orderCount: number
    revenue: number
}
